services-camel/src/main/java/edu/unc/lib/dl/services/camel/images/AddDerivativeProcessor.java

```diff
@@ -48,61 +49,42 @@
 public class AddDerivativeProcessor implements Processor {
     private static final Logger log = LoggerFactory.getLogger(AddDerivativeProcessor.class);
 
-    private final RepositoryObjectLoader repoObjLoader;
-    private final String slug;
     private final String fileExtension;
-    private final String mimetype;
-
-    private final int maxRetries;
-    private final long retryDelay;
+    private final String derivativeBasePath;
 
-    public AddDerivativeProcessor(RepositoryObjectLoader repoObjLoader, String slug, String fileExtension,
-            String mimetype, int maxRetries, long retryDelay) {
-        this.repoObjLoader = repoObjLoader;
-        this.slug = slug;
+    public AddDerivativeProcessor(String fileExtension, String derivativeBasePath) {
         this.fileExtension = fileExtension;
-        this.maxRetries = maxRetries;
-        this.retryDelay = retryDelay;
-        this.mimetype = mimetype;
+        this.derivativeBasePath = derivativeBasePath;
     }
 
     @Override
     public void process(Exchange exchange) throws Exception {
         Message in = exchange.getIn();
         String binaryUri = (String) in.getHeader(FCREPO_URI);
-        int retryAttempt = 0;
+        String binaryId = PIDs.get((String) in.getHeader(CdrBinaryUri)).getId();
```

Comment:
Hmm, CdrBinaryUri is the header that stores the local file path to the original file stored by fedora isn't it? So the file name is actually the checksum of the file rather than the identifier. I think you want to be using binaryUri from line 63 for getting the binaryId instead of CdrBinaryUri